src/membership_check.rs

```diff
@@ -49,10 +55,13 @@ pub async fn membership_check(mut req: AppRequest) -> tide::Result<Response> {
         StatusCode::Ok => {
             let mc_json: MailchimpResponse = mailchimp_res.body_json().await?;
 
-            let membership = if mc_json.status == "pending" || mc_json.status == "subscribed" {
-                "active"
-            } else {
-                "expired"
+            // Parse the expiration date from the MailChimp response
+            let expires_date = parse_iso_date(&mc_json.merge_fields.expires);
+            let today = Local::now().with_timezone(&Vancouver).date_naive();
```

Comment:
@glindstr the signup uses UTC times/dates, not Local. https://github.com/squamishaccess/squamishaccess-functions/blob/main/src/ipn_handler.rs#L232-L233

You should not need `chrono_tz` since we are using UTC.

Reply:
Interesting. I suggest we use pacific time for both calculations. A member who expires on January 1st, 2025 and submits a membership check at 9:00pm pacific prevailing time on December 31st, 2024 will receive an email saying their membership has expired but an expiry date of the next day. A similar situation exists when they sign up or renew they would get an extra day at 9pm. The alternative would be to include the browser timezone in the api call but I suggest we stick with Squamish's timezone and not use utc to prevent this confusion.

The Mailchimp field for Expires and Joined is a pure date type. And it appears their system accepts and convert the RFC3339 datetime format into a pure date. I suggest we use dates instead. 

Are you onboard to switch from chrono::DateTime to chrono::NaiveDate for these two fields? This would require using chrono_tz.

Reply:
If possible I would suggest converting to pacific time (or a localized timezone) when we send the email, to avoid problems with DST. This could be confusing for an admin who looks at mailchimp, I suppose, if they don't know. I think it is still better to store it in UTC in then convert it but if you feel strongly otherwise it can be changed.

Reply:
Using dates is fine but it's less standardized (because timezones), so most storage representations would still be something like RFC 3339.

Reply:
I agree with UTC always being best when available. In this case I believe the mailchimp system does not store UTC or any time information:

https://mailchimp.com/developer/marketing/docs/merge-fields/#structure

The object type is (naive) date and I believe when we send an RFC 3339 timestamp their system simply ignores all time/timezone information.

Reply:
I may be recollecting wrong but it may be that you still need to send an ISO date to the API.

Ok well in that case we should probably send pacific time from the ipn handler function and read it as pacific as well, and just ignore the existence of DST

Reply:
The code does have a comment that there was a requirement for sending datetimes. I believe my other project is sending and receiving naive Dates as strings in json. I'll double check before proceeding. I don't believe we have a  DST issue. Let's see what I come back with and discuss.